src/components/ExportGithub.tsx

```diff
@@ -0,0 +1,211 @@
+  const [repos, setRepos] = useState<UserRepos>(initialRepos);
+  const showGitHubLogin = isGitHubAppInstalled && !gitHubAccessToken;
+  const repoSelectEnabled = isGitHubAppInstalled && gitHubAccessToken;
+
+  const setRepo = (repo: string) => {
+    setSelectedRepo(repo);
+    localStorageSet("selected-repo", repo);
+  };
+
+  const clearAuth = () => {
+    setGitHubAccessToken(null);
+    localStorageSet("oauth-github", "");
+  };
+
+  const fetchAndSetInstallation = async () => {
+    try {
+      const res = await apiGet<{ installations: { app_id: number }[] }>({
+        domain: "https://api.github.com",
+        path: "user/installations",
+        headers: {
+          Authorization: `token ${gitHubAccessToken}`,
+        },
+      });
+      const installations = res.installations;
+      const isAppInstalled = installations.some(
+        (installation) => installation.app_id === 874256
+      );
+      setIsGitHubAppInstalled(isAppInstalled);
+      return isAppInstalled;
+    } catch (error) {
+      const e = error as Error;
+      if (e.message === "Bad credentials") {
+        clearAuth();
+      }
+      return false;
+    }
+  };
+
+  // listen for messages from the auth window
+  useEffect(() => {
+    const handleGitHubAuthMessage = (event: MessageEvent) => {
+      if (event.data) {
+        setGitHubAccessToken(event.data);
+        setClickedInstall(false);
+        localStorageSet("oauth-github", event.data);
+        authWindow.current?.close();
+      }
+    };
+
+    if (isVisible) {
+      window.addEventListener("message", handleGitHubAuthMessage);
+    }
+
+    return () => {
+      if (isVisible) {
+        window.removeEventListener("message", handleGitHubAuthMessage);
+      }
+    };
+  }, [isVisible]);
+
+  // check for installation
+  useEffect(() => {
+    if (gitHubAccessToken) fetchAndSetInstallation();
+  }, [gitHubAccessToken]);
+
+  // get the list of repos
+  useEffect(() => {
+    if (!gitHubAccessToken || !isGitHubAppInstalled) return;
+    const fetchAndSetRepos = async () => {
+      try {
+        const res = await apiGet<UserReposResponse>({
+          domain: "https://api.github.com",
+          path: "user/repos?per_page=100&type=owner",
+          headers: {
+            Authorization: `token ${gitHubAccessToken}`,
+          },
+        });
+        setError("");
+        setRepos(res.data);
+      } catch (error) {
+        setError("Failed to fetch repositories");
+      }
+    };
+    fetchAndSetRepos();
+  }, [gitHubAccessToken, isGitHubAppInstalled]);
+
+  // gatekeep export button
+  useEffect(() => {
+    if (gitHubAccessToken && isGitHubAppInstalled && selectedRepo) {
+      setCanSendToGitHub(true);
+    }
+  }, [gitHubAccessToken, isGitHubAppInstalled, selectedRepo]);
+
+  // clear github token if error writing to github
+  useEffect(() => {
+    if (clearGitHubToken) clearAuth();
+  }, [clearGitHubToken]);
+
+  const InstallButton = () => (
+    <Button
```

Comment:
[nit] We lose some benefits of React's memoization when we define these as new functions each time instead of having them as inline components below. could be further memoized with useCallbacks for each of the onClicks